src/t8_forest/t8_forest_subelements.h

```diff
@@ -0,0 +1,51 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element classes in parallel.
+
+  Copyright (C) 2015 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+/** \file t8_forest_eliminate_hanging_nodes.h
```

Comment:
done, the file is renamed to t8_forest_remove_hanging_nodes.h (and .cxx as well). But from my current point of view all the name changes depend on the decision, whether we implement the subelements as general as possible with removing-hanging-nodes as a special case or if we concentrade on the subelement-removing subelements for now. In this case it would make sense to change all names to something remove-hanging-nodes related. 

As already stated in some other comments, I would preffer a more general implementation (the advanteges are listed in some other comment and you have probably read them by now).

Reply:
The name is fine now since I will focus on removing hanging nodes (faces)